arch/arm/src/common/Toolchain.defs

```diff
@@ -295,6 +295,10 @@ ARCHCXXFLAGS += -nostdinc++ -Wno-attributes -Wno-unknown-pragmas
 # we can safely disregard warnings of the type "parameter passing for X changed in GCC 7.1."
 # Refer to : https://stackoverflow.com/questions/48149323/what-does-the-gcc-warning-project-parameter-passing-for-x-changed-in-gcc-7-1-m
 
+ifeq ($(CONFIG_GLIBCXX),y)
+  ARCHCXXFLAGS :=$(filter-out -nostdinc++,$(ARCHCXXFLAGS))
```

Comment:
let's skip the addition of `-nostdinc` in the first place by checking CONFIG_GLIBCXX